experiment_helper/data.py

```diff
@@ -1,66 +1,56 @@
+from enum import Enum
+
 import torch
-from torch.utils.data.dataset import Subset
+from torch.utils.data.dataset import Subset, Dataset
 import torchvision
 import torchvision.transforms as transforms
-from datasets import load_dataset
+from datasets import load_dataset as huggingface_load_dataset
+
+from pydantic import Field, AliasChoices
+from pydantic_settings import BaseSettings, CliImplicitFlag
 
-from cezo_fl.fl_helpers import get_client_name
 from cezo_fl.util.data_split import dirichlet_split
-from cezo_fl.util.dataset import ShakeSpeare
 from cezo_fl.util.language_utils import (
     LM_DATASET_MAP,
     LM_TEMPLATE_MAP,
-    SUPPORTED_LLM,
     CustomLMDataset,
     CustomLMGenerationDataset,
-    LmTask,
+    LmClassificationTask,
+    LmGenerationTask,
     get_collate_fn,
     get_collate_fn_for_gen_model,
     get_hf_tokenizer,
 )
 
 
-def use_device(args) -> tuple[dict[str, torch.device], dict]:
-    num_clients = args.num_clients
-    use_cuda = not args.no_cuda and torch.cuda.is_available()
-    use_mps = not args.no_mps and torch.backends.mps.is_available()
-    if use_cuda:
-        num_gpu = torch.cuda.device_count()
-        print(f"----- Using cuda count: {num_gpu} -----")
-        # num_workers will make dataloader very slow especially when number clients is large
-        # Do not shuffle shakespeare
-        kwargs = {"pin_memory": True, "shuffle": args.dataset != "shakespeare"}
-        server_device = {"server": torch.device("cuda:0")}
-        client_devices = {
-            get_client_name(i): torch.device(f"cuda:{(i+1) % num_gpu}") for i in range(num_clients)
-        }
-    elif use_mps:
-        print("----- Using mps -----")
-        print("----- Forcing model_dtype = float32 -----")
-        args.model_dtype = "float32"
-        kwargs = {}
-        server_device = {"server": torch.device("mps")}
-        client_devices = {get_client_name(i): torch.device("mps") for i in range(num_clients)}
-    else:
-        print("----- Using cpu -----")
-        print("----- Forcing model_dtype = float32 -----")
-        args.model_dtype = "float32"
-        kwargs = {}
-        server_device = {"server": torch.device("cpu")}
-        client_devices = {get_client_name(i): torch.device("cpu") for i in range(num_clients)}
+class ImageClassificationTask(Enum):
+    mnist = "mnist"
+    cifar10 = "cifar10"
+    fashion = "fashion"
+
 
-    return server_device | client_devices, kwargs
+class DataSetting(BaseSettings, cli_parse_args=True):
```

Comment:
so we may have multiple flags that do not come from the same file, right?

Reply:
this one is special, I am also not sure if we should move this one to the central cli_parser.py. 

I tried to group related logic into same file as much as possible. 
Originally, I wanted to define `class ImageClassificationTask(Enum)` and `get_dataloaders` in this file, and define `DataSetting` in cli_parser.py. But got circular dependency. So I have to choose between 2 solutions:

1. Define `class ImageClassificationTask(Enum)` in a separate file, probably `experiment_typing.py`. Then use it in 2 files.
2. Define everything in this `data.py` file, and import DataSetting in `cli_parser.py` which is what I am doing